drivers/gles3/shaders/scene.glsl

```diff
@@ -2559,22 +2561,51 @@ void main() {
 	diffuse_light *= 1.0 - metallic;
 	vec3 additive_light_color = diffuse_light + specular_light;
 
-#ifndef FOG_DISABLED
+#if !defined(FOG_DISABLED) && !defined(POST_LIGHT_CODE_USED)
 	fog.xy = unpackHalf2x16(fog_rg);
 	fog.zw = unpackHalf2x16(fog_ba);
 
 	additive_light_color *= (1.0 - fog.a);
 #endif // !FOG_DISABLED
 
+#ifndef POST_LIGHT_CODE_USED
 	// Tonemap before writing as we are writing to an sRGB framebuffer
 	additive_light_color *= exposure;
 #ifdef APPLY_TONEMAPPING
 	additive_light_color = apply_tonemapping(additive_light_color, white);
 #endif
 	additive_light_color = linear_to_srgb(additive_light_color);
+#endif
 
 	frag_color.rgb += additive_light_color;
+
 #endif // USE_ADDITIVE_LIGHTING
+
+#if defined(POST_LIGHT_CODE_USED)
+	vec3 out_color = frag_color.rgb;
```

Comment:
This will run once for the base pass, and then once for every additional shadowed light that touches the object

Reply:
Well, that wasn't intended. It would be kind of hard to implement it correctly due to lack of a final pass after all light passes are executed.